django/api/signals.py

```diff
@@ -154,9 +154,13 @@ def create_application(sender, instance, created, **kwargs):
 
 @receiver(post_save, sender=HouseholdMember)
 def after_household_member_save(sender, instance, created, **kwargs):
-    if created and settings.EMAIL["SEND_EMAIL"]:
-        send_household_confirm(
-            recipient=instance.email,
-            id=instance.application.id,
-            initiator_email=instance.application.email,
-        )
+    if created:
+        application = instance.application
+        application.status = GoElectricRebateApplication.Status.SUBMITTED
+        application.save()
```

Comment:
Would this status setting work on a pre-save hook to avoid saving the model a second time?

Reply:
It might be easier though to override the save method in the model itself:

ie https://github.com/django-extensions/django-extensions/blob/main/django_extensions/db/models.py#L20

Reply:
I'm not sure about this...just to clarify, what's happening here is that when a household_member record gets added to the db, the associated primary application has its status updated from "household_initiated" to "submitted". I think this would be the same no matter where we put it - whether in a pre-save or overriding the save() of the HouseholdMember model?

Reply:
Yeah, I think the save method override makes more sense since the logic is part of the model itself. It seems more standard to put something like that there like the above code example. Then another database call is avoided as well.

Reply:
Hmm but I see what you mean now this is the related model, I'm not sure if Django has a cascading save to group queries of related objects into one database call. If it doesn't then we could leave this here in the signal.

Reply:
I tried to test that in a pre_save, but it didn't cascade. I'll try it by overriding save() in HouseholdMember to see if that works.

Reply:
Thanks @tim738745 !!

Reply:
No problem! Unfortunately, it also did not work in save(); even django isn't smart enough to figure out what I want when I do:

    def save(self, *args, **kwargs):
        self.application.status = GoElectricRebateApplication.Status.SUBMITTED
        super().save(*args, **kwargs)

I guess even if it did go through we don't know if, behind the scenes,  django is actually batching the update and insert together...

But anyways I'll just leave signals.py alone and I'll make the change to the serializer.

Reply:
@naomiaro I made the changes and I rebased; should be good now!